asynch/proto/connection.py

```diff
@@ -315,34 +315,27 @@ async def ping(self) -> bool:
             while packet_type == ServerPacket.PROGRESS:
                 await self.receive_progress()
                 packet_type = await self.reader.read_varint()
-            if packet_type is None:
-                logger.debug("Connection closed by remote")
-                await self.disconnect()
-                return False
-            elif packet_type != ServerPacket.PONG:
+            if packet_type != ServerPacket.PONG:
                 msg = self.unexpected_packet_message("Pong", packet_type)
                 raise UnexpectedPacketFromServerError(msg)
             return True
         except AttributeError:
             logger.debug("The connection %s is not open", self)
-            return False
         except IndexError as e:
             logger.debug(
                 "Ping package smaller than expected or empty. "
                 "There may be connection or network problems - "
                 "we believe that the connection is incorrect.",
                 exc_info=e,
             )
-            return False
         except (ConnectionError, OSError, RuntimeError) as e:
             # If raised RuntimeError with "TCPTransport the handler is closed" - just returning false,
             # because this is a connection loss case
             if isinstance(e, RuntimeError) and "TCPTransport closed=True" not in str(e):
                 raise e
-            if isinstance(e, ConnectionError):
-                self.connected = False
+            self.connected = False
```

Comment:
Marking connection closed anyway -> seems good or has potential traps?

Reply:
I wasn't sure about this, as I don't know about the behaviour in case of OSError and RuntimeError, and didn't feel comfortable about changing this without being able to reproduce these errors.

Reply:
Hm, if a socket is closed ( https://github.com/long2ice/asynch/pull/129/files#diff-73d396656d1c5395ea13dc77d4892584ed77d53d27c815212e44b72b7366de98R344 ), then the connection may be not technically closed, but inoperative.

The main question: should `ping` also (=and) mark the connection closed? I think no because it violates SRE principle in a way, so we can remove this line at all, but no pressure, let's have a discussion.